static/app/components/events/interfaces/sourceMapsDebuggerModal.tsx

```diff
@@ -43,16 +45,27 @@ interface SourceResolutionResults {
     | {status: 'none'};
   stackFrameDebugId: string | null;
   stackFramePath: string | null;
-  uploadedSomeArtifact: boolean;
-  uploadedSomeArtifactToRelease: boolean;
   uploadedSomeArtifactWithDebugId: boolean;
   uploadedSourceFileWithCorrectDebugId: boolean;
   uploadedSourceMapWithCorrectDebugId: boolean;
-  sdkVersion?: string;
 }
 
 interface SourceMapsDebuggerModalProps extends ModalRenderProps {
-  sourceResolutionResults: SourceResolutionResults;
+  sourceResolutionResults: FrameSourceMapDebuggerData;
+}
+
+export function frameIsFullyResolvedBasedOnDebuggerData(
+  frameDebuggerData: FrameSourceMapDebuggerData
+): boolean {
+  const {debugIdProgressPercent} = getDebugIdProgress(frameDebuggerData);
```

Comment:
l: I don't like that we have to run the getters to decide if we have to render the component, and then run it again when the component renders.

Can we lift this logic out of the `SourceMapsDebuggerModal` and inject in the progress percent as props?

There's a bunch of props to inject so we could also inject it all as a single object.

Reply:
The progress could already be calculated in `mapSourceMapDebuggerFrameInformation`